pkg/controller/sparkapplication/controller.go

```diff
@@ -672,9 +683,54 @@ func (c *Controller) submitSparkApplication(app *v1beta1.SparkApplication) *v1be
 			}
 		}
 	}
+
+	//Use PodGroup to schedule spark application executor pods if required
+	if c.volcanoClient != nil && app.VolcanoSchedulingRequired() {
+		if app.Spec.Executor.Annotations == nil {
+			app.Spec.Executor.Annotations = make(map[string]string)
+		}
+		if _, ok := app.Spec.Executor.Annotations["scheduling.k8s.io/group-name"]; !ok {
```

Comment:
I'm thinking to also include driver into PodGroup.

Reply:
The problem is the executors will be created only after the driver is started. Does `PodGroup` works in this case?

Reply:
yes, podgroup can help on that: set `minNumber` to `1` to start driver, and set `minResoruce` to other value, e,g. 1 driver with 3 executors; the admission of Volcano will only admit the `PodGroup`s that total of `minResource` is less than `cluster resource * threshold (1.2 by default)`. That'll help to reducing the chance that driver pod consume so many resources :)

Reply:
Other than that, will Valcano guarantee that it will only admit a `PodGroup` when there exists enough resources in the cluster to run the `minSize` number of pods in the group? If not, the value of creating a `PodGroup` that includes the driver pod is much less. And will Valcano reserve the needed resources to run the pods?

Reply:
> Other than that, will Valcano guarantee that it will only admit a PodGroup when there exists enough resources in the cluster to run the `minSize` number of pods in the group? 

Almost yes :) If `total minSize` is larger than `cluster resource * threshold`, both driver and executor pod will not admit. Threshold was introduced because admission does not mean schedulable, e.g. node selector; it give a balance between admission and schedulable.

> And will Valcano reserve the needed resources to run the pods?

hm.... we don-t reserve resources but we will schedule next job until `minResources` was fitted.